### ISSUE TITLE
about training code 

### ISSUE THREAD

**Issue by bighan123** [YuYue525/MobileSAM-pytorch#13]
You did great work!!! but may I ask if you used knowledge distillation during the training? Because in the training code I only see the tinyvit model, and the mobilesam article mentions that they distill SAM

**Comment by YuYue525** [comment]
We follows the implementation of MobileSAM. We regard the encoder feature output of SAM as the target "knowledge" and do distillation. Actually, in the repo of MobileSAM, the prompt encoder & mask decoder's weights are the same as the SAM's, they didn't update the weights of prompt encoder & mask decoder as well.

**Comment by bighan123** [comment]
Thanks for responsing to this, does this mean that the model is only supervised by the teacher model (SAM)? Because the loss during training only contains target_feats and pred_feats, and targer_feats are the features encoded by the SAM encoder

**Comment by YuYue525** [comment]
Yes. To be precise, the lightweight encoder TinyViT (student model) is only supervised by the SAM encoder (teacher model). 
In the paper, the author also mentioned that we can retrain the mask decoder as well, but some people have tried this, which will reduce the generalization ability of the model.

**Comment by bighan123** [comment]
Thank you very much for your help!!! by the way, does saving sam's coded features of the image just save time in forward propagation? Does it help the model performance?

**Comment by YuYue525** [comment]
It saves time because it's unnecessary for the original SAM encoder to do forward propagation during training. By the way, make sure your hard drive have enough space to save SAM's coded features. 
This does not affect model performance.
